### PR TITLE
Fix amq router race condition

### PR DESCRIPTION
**Pull request by jzding**
Add retries when starting AMQP server to deal with the race condition when AMQP router starts later than cloud-event-proxy service, for example when a node is rebooted.

Timeout for retry duration is set to 30 seconds.

Signed-off-by: Jack Ding <jackding@gmail.com>

### CODE REVIEW

File: v1/amqp/amqp.go [hunk 1]
```diff
@@ -15,39 +15,58 @@
...
+		}
+
+		instance = &AMQP{
+			Router: router,
+		}
+		break
+	}
 	if instance == nil || instance.Router == nil {
-		return nil, errorhandler.AMQPConnectionError{Desc: "amqp connection error"}
+		return nil, errorhandler.AMQPConnectionError{}
```
Comment (aneeshkp) [inline v1/amqp/amqp.go:67]: why removed description ?

Comment (aneeshkp) [reply v1/amqp/amqp.go:67]: text might be used for parsing some where

Comment (jzding) [reply v1/amqp/amqp.go:67]: The `AMQPConnectionError` itself already has description.
Before removing description:

> ERRO[0000] error starting amqp at amqp:localhost:5672 error: amqp connection error amqp connection error 
> WARN[0000] requires QPID router installed to function fully amqp connection error amqp connection error 

After removing description:

> ERRO[0000] error starting amqp at amqp:localhost:5672 error: amqp connection error  
> WARN[0000] requires QPID router installed to function fully amqp connection error

Comment (aneeshkp) [reply v1/amqp/amqp.go:67]: nice

**Review by aneeshkp — APPROVED** [review]

**Pull request merged**